packages/data/src/plugins/actions.js

```diff
@@ -125,6 +126,62 @@ export function* installAndActivatePlugins( plugins ) {
 	}
 }
 
+export const createErrorNotice = ( errorMessage ) => {
+	return dispatch( 'core/notices', 'createNotice', errorMessage );
+};
+
+export function* installJetpackAndConnect() {
+	try {
+		yield dispatch( STORE_NAME, 'installPlugins', [ 'jetpack' ] );
+		yield dispatch( STORE_NAME, 'activatePlugins', [ 'jetpack' ] );
+
+		const url = yield select( STORE_NAME, 'getJetpackConnectUrl', {
+			redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
+		} );
+
+		// getJetpackConnectUrl doesn't throw errors, it sets error state if
```

Comment:
My personal preference on this is that we do both (throw errors and set them in state for later handling) to allow consumers to easily access this data.

Without throwing errors, we can create race conditions when error checking in components.

But would love to hear others thoughts on this and it's something we can tackle in a follow-up.

Reply:
@joshuatf yes this was something that felt a little out of scope for the refactor since the existing functionality does not throw in many cases and instead puts errors in state. I think we're in agreeance that a pattern of throwing makes much more sense and I would want to follow up in the future and refactor out this pattern.

Reply:
If you're happy to push this out to future refactors I think thats what I'd want to do here. Let me know what you think.

Reply:
Absolutely- we can tackle these in a follow-up and make them more consistent across all our data packages.

Reply:
Created a follow-up issue to take care of this in all data stores here - https://github.com/woocommerce/woocommerce-admin/issues/4963

Reply:
Thanks!